### DISCUSSION TITLE
Register assets context

### DISCUSSION THREAD

**Discussion by JoryHogeveen** [pods-framework/pods#6795] [Ideas]
Pods hooks into all register script hooks of WordPress. 2.7.23 will add two filters (before and after) for other parts of Pods to register script.
We should add some form of context to these filters. Some ideas:

- `admin` General admin pages
- `front` Frontend pages
- `login` Login page
- `builder` Page Builder pages (usually frontend)
- `modal` Modal view (`pods_is_modal_window()`), this is similar to admin context

[a comment by an automated account is not shown]

**Comment by sc0ttkclark** [comment]
@JoryHogeveen Can you clarify what hooks / context would be used in an example?

**Comment by JoryHogeveen** [comment]
Sure!

There are now two filters: `pods_before_enqueue_scripts` and `pods_after_enqueue_scripts`

We could add a second context parameter. Example of usage:

```php
add_filter( 'pods_after_enqueue_scripts', function( $suffix, $context ) {
    if ( 'admin' === $context ) {
        // Scripts for admin pages only.
    }

    if ( 'modal' === $context ) {
        // Scripts for Pods modal only.
    }

    // etc.
}, 10, 2 );
```

I already used the filter in PodsI18n: https://github.com/pods-framework/pods/blob/main/classes/PodsI18n.php#L39

**Comment by lkraav** [comment]
To avoid reinventing wheels, may I suggest taking a look at what our Inpsyde buddies have thought of this problem at https://github.com/inpsyde/wp-context/

**Comment by JoryHogeveen** [comment]
Hi @lkraav 
The only problem with that package is that is isn't extendable with filters to add custom contexts as far as I can see.

**Comment by lkraav** [comment]
> The only problem with that package is that is isn't extendable with filters to add custom contexts as far as I can see.

Possibly, I'm in the beggining phases with it myself, through another Inpsyde package I'm using.

But this maybe begs the questions, how many context should be defined, and if we think we need more, is there a problem with modelling our problem?

**Comment by JoryHogeveen** [comment]
For Pods we have the `modal` and `builder` context as well.

Modal: Pods modal popup
Builder: Frontend layout with page builder active

These are both quite important contexts in this case but are not part of WP core.
